libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline.rs

```diff
@@ -42,12 +42,13 @@ impl<'a, 'b> ValidationPipeline<'a> {
         // Phase 1 is source block loading.
 
         // Phase 2: Name resolution.
-        let db = ParserDatabase::new(ast_schema, &mut diagnostics);
+        let db = match ParserDatabase::new(ast_schema, &mut diagnostics) {
```

Comment:
This is quite weird, to return `None` if we cannot create a database. And then returning the diagnostics...

Reply:
Good point, I think we could make this never fail instead, that would be more consistent.